build.rs

```diff
@@ -6,19 +6,37 @@ fn main() {
     let manifest_dir = PathBuf::from(env::var_os("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR"));
     let vendor = manifest_dir.join("vendor");
 
+    if !vendor.join("src").exists() {
+        panic!("{} dir is missing files. Try running: `git submodule update --init --recursive`", vendor.display());
+    }
+
     let mut cc = cc::Build::new();
     cc.include(&vendor);
     cc.define("NDEBUG", Some("1"));
     cc.define("_THREAD_SAFE", Some("1"));
-    cc.flag("-fvisibility=hidden"); // FIXME: msvc?
+    if !cc.get_compiler().is_like_msvc() {
+        cc.flag("-fvisibility=hidden");
+    }
+
+    if let Ok(target_cpu) = env::var("TARGET_CPU") {
+        cc.flag_if_supported(&format!("-march={}", target_cpu));
+    }
 
     let target = env::var("CARGO_CFG_TARGET_ARCH").expect("CARGO_CFG_TARGET_ARCH");
     match target.as_str() {
         "x86_64" | "i686" => {
             cc.define("WEBP_HAVE_SSE2", Some("1"));
+            if cfg!(feature = "sse41") {
```

Comment:
There's `#[target_feature(enable = "sse4.1")]`, any reason to use regular features instead?

Reply:
Yes, because machine that compiles this code may have a different CPU than the machine which is meant to run it.

Reply:
Aren't feature flags controlled by `target-cpu` in `rustflags`? If you compile for some other CPU, you'll get feature flags automatically set accordingly.